massa-execution/src/interface_impl.rs

```diff
@@ -271,4 +343,23 @@ impl Interface for InterfaceImpl {
         // TODO store the event somewhere
         Ok(())
     }
+
+    /// Returns the current time (millisecond unix timestamp)
+    fn get_time(&self) -> Result<u64> {
+        let context = context_guard!(self);
+        let ts = get_block_slot_timestamp(
+            self.thread_count,
+            self.t0,
+            self.genesis_timestamp,
+            context.slot,
+        )?;
+        Ok(ts.to_millis())
+    }
+
+    /// Returns a random number (unsafe: can be predicted and manipulated)
+    fn unsafe_random(&self) -> Result<i64> {
+        let mut context = context_guard!(self);
```

Comment:
Same here, you could first comput `distr`, then acquire the guard and use it.

Reply:
done